src/Analyser/NodeScopeResolver.php

```diff
@@ -823,6 +823,9 @@ private function processStmtNode(
 			if (!$isIterableAtLeastOnce->no()) {
 				$throwPoints = array_merge($throwPoints, $finalScopeResult->getThrowPoints());
 			}
+			if (!$scope->getType($stmt->expr)->isArray()->yes()) {
```

Comment:
What about this instead?

```php
!(new ObjectType(\ArrayAccess))->isSuperTypeOf($scope->getType($stmt->expr)-)->no()
```

Reply:
This will make it not report a false positive for example `array|null`.

Reply:
Thank you! I'll look into it.